src/lib.rs

```diff
@@ -266,18 +266,33 @@ pub mod util {
     }
 
     pub fn load_private_key(filename: &str) -> Result<rustls::PrivateKey> {
-        let keyfile = fs::File::open(filename).map_err(|e| Error::Io(e))?;
+        use std::io::Seek;
+        use std::io::BufRead;
+
+        let keyfile = fs::File::open(filename).map_err(Error::Io)?;
         let mut reader = BufReader::new(keyfile);
-        let mut keys = pemfile::rsa_private_keys(&mut reader)
-            .map_err(|_| Error::BadKey)?;
 
-        // Ensure there's only one key in the file.
-        if keys.len() != 1 {
-            return Err(Error::BadKeyCount);
-        }
+        let private_keys_fn = {
```

Comment:
Can you add a comment about why this is happening here, for future reference? Just something short and pithy, nor more than a couple of lines.